cli/src/main/java/com/devonfw/tools/ide/tool/kubectl/KubeCtl.java

```diff
@@ -0,0 +1,28 @@
+package com.devonfw.tools.ide.tool.kubectl;
+
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.process.EnvironmentContext;
+import com.devonfw.tools.ide.tool.GlobalToolCommandlet;
+import com.devonfw.tools.ide.tool.docker.Docker;
+
+public class KubeCtl extends GlobalToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public KubeCtl(IdeContext context) {
+
+    super(context, "kubectl", Set.of(Tag.DOCKER));
+  }
+
+  @Override
+  public boolean install(boolean silent, EnvironmentContext environmentContext) {
+    // TODO create kubectl/kubectl/dependencies.json file in ide-urls and delete this method
+    return getCommandlet(Docker.class).install(silent, environmentContext);
+  }
```

Comment:
We will have the same pattern multiple times.
Either we consider something like this as a proxy commandlet (create abstract class?) and just define the actual delegate commandlet via some protected method that we override or we would override a protected method like `protected boolean skipOwnInstallation()`. IMHO we should discuss this with the team in a next meeting.

Reply:
We can agree that here is a case where we do not need `dependencies.json` and a hardcoded dependency makes sense. There is no kubectl in ide-urls and there is no need to ever add it. This is and will always be a commandlet delegating to `Docker`.
Did you already try to implement such abstract class `DelegatingCommandlet` (if you think `ProxyCommandlet` might be confusing since it has nothing to do with an HTTP proxy)?

Reply:
Yes, i already implemented a ProxyCommandlet but i am not sure if i have done so correctly. I would like to discuss it in the next meeting